### PR TITLE
INSTUI-3467 - generate table of content to docs pages

### PR DESCRIPTION
**Pull request by ke1k0**

### CODE REVIEW

**Review by Brailor — APPROVED** [review]
This is great!
I would prefer the TOC to be on the right side of the page, but this is just a personal preference.

**Comment by ke1k0** [comment]
> This is great! I would prefer the TOC to be on the right/left side of the page and ideally it would be "sticky", so no matter how scrolled down are you on the page it would be always visible, but this is just a personal preference.

@Brailor Yes indeed, that would be better. The problem is that the page already has a side menu, and another one on the right side would take too much space away from the content. I consulted with Ákos about this, and he agreed that it wouldn't work well on smaller viewports either, and might even have some a11y issues. We talked about redesigning the whole docs page in the future, but in the meantime we agreed to place the PoC on the top of the page.

**Comment by matyasf** [comment]
I'd leave out Notes and warnings, they look odd in the TOC, see for example https://996--preview-instui.netlify.app/#using-theme-overrides https://996--preview-instui.netlify.app/#server-side-rendering 

Also maybe dont display the TOC if it has less than 4 items or the page is less than say 1500 characters? it does not look nice e.g. in the Packages category https://996--preview-instui.netlify.app/#babel-plugin-transform-imports

**Review by matyasf — CHANGES_REQUESTED** [review]
Code looks good, I just left a comment about how it should work in some cases

File: packages/__docs__/src/Document/index.tsx [hunk 1]
```diff
@@ -291,8 +297,105 @@ import { ${importName} } from '${esPath}'
...
+  //             size="small"
+  //             itemSpacing="xx-small"
+  //           >
+  //             {TOC}
+  //           </List>
+  //         </ToggleDetails>
+  //       </View>
+  //     </InstUISettingsProvider>
+  //   )
+  // }
```
Comment (matyasf) [inline packages/__docs__/src/Document/index.tsx:394]: You can delete this :)

**Review by matyasf — APPROVED** [review]
looks good except that large commented out code can be deleted

File: packages/__docs__/src/Document/index.tsx (hunk 1 above)
Comment (ke1k0) [reply packages/__docs__/src/Document/index.tsx]: Oh boy, forgot to clean up! :D Thanks :)

**Pull request merged**